blocks/library/table2/index.js

```diff
@@ -0,0 +1,98 @@
+/**
+ * Internal dependencies
+ */
+import './style.scss';
+import { registerBlock, query as hpq } from '../../api';
+import TableBlock from './table-block';
+
+const { children } = hpq;
+
+/**
+ * Returns an attribute setter with behavior that if the target value is
+ * already the assigned attribute value, it will be set to undefined.
+ *
+ * @param  {string}   align Alignment value
+ * @return {Function}       Attribute setter
+ */
+function toggleAlignment( align ) {
+	return ( attributes, setAttributes ) => {
+		const nextAlign = attributes.align === align ? undefined : align;
+		setAttributes( { align: nextAlign } );
+	};
+}
+
+registerBlock( 'core/table2', {
+	title: wp.i18n.__( 'Table2' ),
+	icon: 'editor-table',
+	category: 'common',
+
+	attributes: {
+		content: children( 'table' ),
+	},
+
+	defaultAttributes: {
+		content: [
+			<tbody key="1">
```

Comment:
We don't need to address it now, but there's some related chatter in #892 about how to handle or do away with attributes defaulting during initialization. Feel free to leave feedback there if you have thoughts, otherwise we can encounter this when I circle back to that pull request.